charts/consul/templates/cni-clusterrole.yaml

```diff
@@ -0,0 +1,24 @@
+{{- if (and (.Values.connectInject.cni.enabled) (not .Values.connectInject.transparentProxy.defaultEnabled)) }}{{ fail "connectInject.transparentProxy.defaultEnabled must be true if connectInject.cni.enabled is true" }}{{ end -}}
+{{- if (and .Values.connectInject.cni.enabled .Values.connectInject.transparentProxy.defaultEnabled) }}
```

Comment:
I *think* this second line is redundant? On line 1 we fail if `cni is enabled && tproxy is disabled`

Reply:
nevermind! this would otherwise render always! :)

Reply:
No, I don't thinks so.  

Throw an error if cni.enabled=true and transparentProxy.defaultEnabled=false

If both cni.enabled=true and transparentProxy.defaultEnabled=true continue on.